### ISSUE TITLE
ksql-cli hangs after server maintenance, WARN No records received after 30 seconds of polling in log

### ISSUE THREAD

**Issue by alexhafner** [confluentinc/ksql#303]
We had to cycle our ksql dev box. Sadly KSQL hadn't been played with for a few days; however after the server maintenance we attempted to go back into the CLI in local mode (non-dockerized) which did not show any response or error after launching it.

The CLI started to repeat

`[2017-09-21 15:17:33,781] WARN No records received after 30 seconds of polling; something may be wrong (io.confluent.ksql.rest.server.computation.CommandStore:153)`

After updating our build to the latest release the issue persisted. We then deleted test topics in this order

ksql_transient*
```
ksql_transient_528162554338760763_1505245884471-TEST_Atransient_-changelog
ksql_transient_528162554338760763_1505245884471-TEST_Atransient_-repartition
ksql_transient_6259632277308348632_1505245661426-TEST_Atransient_-changelog
ksql_transient_6259632277308348632_1505245661426-TEST_Atransient_-repartition
ksql_transient_7529857592400492031_1505246139152-TEST_Atransient_-changelog
ksql_transient_7529857592400492031_1505246139152-TEST_Atransient_-repartition
```
ksql_query_1-KSQL_Agg_Query*
```
ksql_query_1-KSQL_Agg_Query_1505245639135-changelog	
ksql_query_1-KSQL_Agg_Query_1505245639135-repartition	
```
and finally 
`ksql__commands
`

The CLI was stopped and started after each of the 3 groups of topics. After deleting ksql_commands (the topic was empty) and restarting the CLI we got to the Welcome Screen and command prompt again. 

A full log file is here [ksql_hung.txt](https://github.com/confluentinc/ksql/files/1321964/ksql_hung.txt)

I could not reproduce the issue after that, will add info if a pattern emerges.

**Comment by bluemonk3y** [comment]
Hi Victor,
I believe we have seen something similar before. It would be useful to capture the thread dump to see that threads blocked causing the hang. I guess you know already but you can use jps to identify the process and then jstack to get a thread dump. 
Thanks Neil.

**Comment by alexhafner** [comment]
Hi Neil - OK we'll capture that next time we run into that. Cheers Alex

**Comment by geojub** [comment]
Hi all,

Yesterday I've experienced the same issue as @alexhafner , "ksql-cli local" worked very well when I first used it and after that it started hanging. I followed @alexhafner  suggestions erasing the transient and the queries files, I even recompiled ksql again, but with no luck.

I have ksql installed also on other two systems and it works just fine. I couldn't replicate the same issue on these systems.

I will attach the thread dump and the log file. 
[ksql-cli-local-hang.log](https://github.com/confluentinc/ksql/files/1335114/ksql-cli-local-hang.log)

[ksql-cli-local-process-dump.log](https://github.com/confluentinc/ksql/files/1335116/ksql-cli-local-process-dump.log)
